tf_agents/agents/dqn/rnddqn_agent.py

```diff
@@ -0,0 +1,456 @@
+      td_errors_loss_fn: A function(td_targets, predictions) to compute the
+        element wise loss.
+      rnd_loss_fn: A function(rnd_targets, predictions) to compute the
+        element wise loss.
+      gamma: Discount for future rewards.
+      reward_scale_factor: Multiplicative factor to scale rewards.
+      weights: Optional scalar or elementwise (per-batch-entry) importance
+        weights.  The output td_loss will be scaled by these weights, and
+        the final scalar loss is the mean of these values.
+
+    Returns:
+      loss: An instance of `DqnLossInfo`.
+    Raises:
+      ValueError:
+        if the number of actions is greater than 1.
+    """
+    # Check that `experience` includes two outer dimensions [B, T, ...]. This
+    # method requires a time dimension to compute the loss properly.
+    self._check_trajectory_dimensions(experience)
+
+    if self._n_step_update == 1:
+      time_steps, actions, next_time_steps = self._experience_to_transitions(
+          experience)
+    else:
+      # To compute n-step returns, we need the first time steps, the first
+      # actions, and the last time steps. Therefore we extract the first and
+      # last transitions from our Trajectory.
+      first_two_steps = tf.nest.map_structure(lambda x: x[:, :2], experience)
+      last_two_steps = tf.nest.map_structure(lambda x: x[:, -2:], experience)
+      time_steps, actions, _ = self._experience_to_transitions(first_two_steps)
+      _, _, next_time_steps = self._experience_to_transitions(last_two_steps)
+
+    # Compute RND loss
+    # RND loss should be calculated first, since it is used as intrinsic rewards
+    with tf.name_scope('rnd_loss'):
+      rnd_prediction, _ = self._rnd_network(time_steps.observation,
+                                            time_steps.step_type)
+      rnd_target, _ = self._target_rnd_network(time_steps.observation,
+                                               time_steps.step_type)
+
+      rnd_losses = element_wise_squared_loss(rnd_prediction, rnd_target)
+
+      # if nest_utils.is_batched_nested_tensors(
+      #     time_steps, self.time_step_spec, num_outer_dims=2):
+      #   # Do a sum over the time dimension.
+      #   td_loss = tf.reduce_sum(input_tensor=td_loss, axis=1)
+
+      # Average across the elements of the batch.
+      # Note: We use an element wise loss above to ensure each element is always
+      #   weighted by 1/N where N is the batch size, even when some of the
+      #   weights are zero due to boundary transitions. Weighting by 1/K where K
+      #   is the actual number of non-zero weight would artificially increase
+      #   their contribution in the loss. Think about what would happen as
+      #   the number of boundary samples increases.
+      avg_rnd_loss = tf.reduce_mean(input_tensor=rnd_losses)
+
+      with tf.name_scope('Losses/'):
+        tf.compat.v2.summary.scalar(
+            name='loss', data=avg_rnd_loss, step=self.train_step_counter)
```

Comment:
```suggestion
            name='rnd_loss', data=avg_rnd_loss, step=self.train_step_counter)
```

Reply:
![Screenshot from 2019-06-30 01-13-01](https://user-images.githubusercontent.com/6107926/60386748-4b627a00-9ad4-11e9-8f99-b95c15b93d94.png)

This was indeed the problem! Thank you Oscar